drivers/base/I2C_Device.zig

```diff
@@ -100,8 +121,10 @@ pub fn readv(dev: I2C_Device, datagrams: []const []u8) InterfaceError!usize {
     return readv_fn(dev.ptr, datagrams);
 }
 
+const AllowReserved = enum { AllowReserved, DontAllowReserved };
```

Comment:
Use `snake_case` here! I would also probably solve the "allow generic call or not" with this?

Reply:
> But for writev we don't want to add that argument, so we'll still have to just explicitly allow it like so:

Why wouldn't we want that argument there? :thinking:

Reply:
I suppose we could, I just don't think that we want to add more to the writev/writev_blocking/write/write_blocking/write_then_read/read/readv etc argument list. I think it's probably in general to just allow general for writing but not reading, since as far as I can tell none of the other types make sense in any other context.

We could maybe make it an option in the hal? so we'd do like `i2c0.apply(.{..., .allow_reserved = .AllowGeneral})`

Reply:
But also that can be figured out later since none of the HALs implement this interface yet.